src/lib.rs

```diff
@@ -83,54 +92,104 @@ mod window_counter;
 pub mod connectors;
 pub mod resolvers;
 
+/// Uniquely identifies a claim
+#[derive(Copy, Clone, Debug)]
+pub(crate) struct ClaimId(pub u64);
+
+impl ClaimId {
+    fn new() -> Self {
+        let id = usdt::UniqueId::new().as_u64();
+        Self(id)
+    }
+}
+
+impl Default for ClaimId {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 /// USDT probes for tracing how qorb makes pools and hands out claims.
 #[cfg(feature = "probes")]
 #[usdt::provider(provider = "qorb")]
 mod probes {
     /// Fires right before attempting to acquire a claim from the pool.
-    fn claim__start(id: &usdt::UniqueId) {}
+    fn claim__start(pool: &str, claim_id: u64) {}
 
     /// Fires when a claim is successfully acquired from the pool.
-    fn claim__done(id: &usdt::UniqueId) {}
+    ///
+    /// Also identifies the underlying slot which is being used.
+    fn claim__done(pool: &str, claim_id: u64, slot_id: u64) {}
 
     /// Fires when we _fail_ to acquire a claim from the pool, with a string
     /// identifying the reason.
-    fn claim__failed(id: &usdt::UniqueId, reason: &str) {}
+    fn claim__failed(pool: &str, claim_id: u64, reason: &str) {}
 
     /// Fires right before attempting to make a connection, with the address
     /// we're connecting to.
-    fn connect__start(id: &usdt::UniqueId, addr: &str) {}
+    fn connect__start(pool: &str, slot_id: u64, addr: &str) {}
 
     /// Fires just after successfully making a connection.
-    fn connect__done(id: &usdt::UniqueId) {}
+    fn connect__done(pool: &str, slot_id: u64, addr: &str) {}
 
     /// Fires just after failing to make a connectiona, with a string
     /// identifying the reason.
-    fn connect__failed(id: &usdt::UniqueId, reason: &str) {}
+    fn connect__failed(pool: &str, slot_id: u64, addr: &str, reason: &str) {}
+
+    /// Fires when the pool is attempting to make a claim.
+    ///
+    /// This issimilar to "claim__start", but it follows the internal
+    /// pool-specific task, which may happen multiple times if e.g.
+    /// waiting for a backend to be ready.
```

Comment:
Can you say more about this?  You'd get multiple `pool-claim-start` probes for the same `claim-start` probe?

Reply:
Sure, I'll update the comment.